File: algorithm/104.二叉树的最大深度.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */

var maxDepth = function(root) {
  if (!root) return 0;
  const leftDeep = maxDepth(root.left);
  const rightDeep = maxDepth(root.right);
  const max = Math.max(leftDeep, rightDeep);
  return max + 1;
};

File: algorithm/104.二叉树的最大深度.py
# Definition for a binary tree node.
class TreeNode:
  def __init__(self, x):
    self.val = x
    self.left = None
    self.right = None

class Solution:
  def maxDepth(self, root):
    if not root: return 0
    queue = [root]
    deep = 1;
    while len(queue):
      queueLen = len(queue)
      while queueLen:
        queueLen -= 1
        temp = queue[0:1][0]
        queue = queue[1:]
        if temp.left: queue.append(temp.left)
        if temp.right: queue.append(temp.right)
      deep += 1
    return deep
